service/fullnode/fullnode/fullnode.go

```diff
@@ -0,0 +1,221 @@
+	bc, err := core.NewBlockChain(&config.ChainConfig, "fullnode")
+	if err != nil {
+		return nil, fmt.Errorf("new blockchain error: %v", err)
+	}
+
+	l2EndPoint := config.ApiServer.L2EndPoint
+	if len(l2EndPoint) == 0 {
+		l2EndPoint = DefaultL2EndPoint
+	}
+
+	if config.FullNode.SyncBlockStatus <= block.StatusProposing ||
+		config.FullNode.SyncBlockStatus > block.StatusVerifiedAndExecuted {
+
+		config.FullNode.SyncBlockStatus = block.StatusVerifiedAndExecuted
+	}
+
+	fullnode := &Fullnode{
+		config: config,
+		client: client.NewZkBNBClient(l2EndPoint),
+		bc:     bc,
+
+		quitCh: make(chan struct{}),
+	}
+	return fullnode, nil
+}
+
+func (c *Fullnode) Run() {
+	// TODO: add BlockSize in zkbnb-go-sdk
+
+	curHeight, err := c.bc.BlockModel.GetCurrentBlockHeight()
+	if err != nil {
+		panic(fmt.Sprintf("get current block height failed, error: %v", err.Error()))
+	}
+
+	curBlock, err := c.bc.BlockModel.GetBlockByHeight(curHeight)
+	if err != nil {
+		panic(fmt.Sprintf("get current block failed, height: %d, error: %v", curHeight, err.Error()))
+	}
+
+	ticker := time.NewTicker(SyncInterval)
+	defer ticker.Stop()
+
+	syncBlockStatus := c.config.FullNode.SyncBlockStatus
+	for {
+		select {
+		case <-ticker.C:
+
+			// if the latest block have been created
+			if curBlock.BlockStatus > block.StatusProposing {
+				// init new block, set curBlock.status to block.StatusProposing
+				curBlock, err = c.bc.InitNewBlock()
+				if err != nil {
+					panic(fmt.Sprintf("init new block failed, block height: %d, error: %v", curHeight, err.Error()))
+				}
+
+				curHeight++
+			}
+
+			l2Block, err := c.client.GetBlockByHeight(curHeight)
+			if err != nil && err != types.DbErrNotFound {
+				logx.Errorf("get block failed, height: %d, err %v ", curHeight, err)
+				continue
+			}
+
+			if l2Block.Status < syncBlockStatus {
+				continue
+			}
+
+			for _, blockTx := range l2Block.Txs {
+				newTx := &tx.Tx{
+					TxHash: blockTx.Hash, // Would be computed in prepare method of executors.
+					TxType: blockTx.Type,
+					TxInfo: blockTx.Info,
+				}
+
+				err = c.bc.ApplyTransaction(newTx)
+				if err != nil {
+					logx.Errorf("apply block tx ID: %d failed, err %v ", newTx.ID, err)
+					continue
+				}
+			}
+
+			curBlock.BlockSize = l2Block.Size
+
+			if c.bc.Statedb.StateRoot != l2Block.StateRoot {
+				panic(fmt.Sprintf("state root not matched between statedb and l2block: %d", l2Block.Height))
+			}
+
+			logx.Infof("created new block on fullnode, height=%d, blockSize=%d", curBlock.BlockHeight, curBlock.BlockSize)
+			curBlock, err = c.processNewBlock(curBlock)
+			if err != nil {
+				panic(fmt.Sprintf("new block failed, block height: %d, Error: %s", l2Block.Height, err.Error()))
+			}
+
+		case <-c.quitCh:
+			return
+		}
+	}
+}
+
+func (c *Fullnode) Shutdown() {
+	close(c.quitCh)
+	c.bc.Statedb.Close()
+	c.bc.ChainDB.Close()
+}
+
+func (c *Fullnode) processNewBlock(curBlock *block.Block) (*block.Block, error) {
```

Comment:
We can make blockSize as an input here, and pass it to `c.bc.CommitNewBlock`, the block size will be set in the `CommitNewBlock`, no need to set it externally.